### PR TITLE
[fx] add a symbolic_trace api.

### PR DESCRIPTION
**Pull request by super-dainiu**
### What's new?
To create a simple API for `symbolic_trace` which is more user-friendly.
https://github.com/pytorch/pytorch/blob/master/torch/fx/_symbolic_trace.py#L1018
Not too much to comment on here, keep everything compact.
```python
# The failure of `ColoTracer` with `meta_args` made our tracing unstable.
>>> model = ...

# if this works
>>> gm = symbolic_trace(model)

# else try this
>>> gm = symbolic_trace(model, meta_args={'x': torch.rand(1, 3, 224, 224, device='meta')})

# else try this
>>> gm = symbolic_trace(model, concrete_args={'x': torch.rand(1, 3, 224, 224)})
```
### Test
image

### CODE REVIEW

File: tests/test_fx/test_tracer/test_torchvision_model/test_torchvision_model.py [hunk 1]
```diff
@@ -30,10 +29,7 @@ def test_torchvision_models():
         else:
             model = model_cls()
 
-        graph = tracer.trace(root=model)
-
-        gm = GraphModule(model, graph, model.__class__.__name__)
-        gm.recompile()
+        gm = symbolic_trace(model)
```
Comment (super-dainiu) [inline tests/test_fx/test_tracer/test_torchvision_model/test_torchvision_model.py:32]: The bad case is we cannot use `ColoTracer` with `meta_args` here.

File: tests/test_fx/test_tracer/test_torchrec_model/test_deepfm_model.py [hunk 2]
```diff
@@ -60,8 +53,7 @@ def test_torchrec_deepfm_models():
             model = model_cls(ebc)
 
         # Setup GraphModule
-        graph = tracer.trace(model)
-        gm = GraphModule(model, graph, model.__class__.__name__)
+        gm = symbolic_trace(model)
         gm.recompile()
```
Comment (YuliangLiu0306) [inline tests/test_fx/test_tracer/test_torchrec_model/test_deepfm_model.py:57]: this recompile call may be redundant.

File: tests/test_fx/test_tracer/test_hf_model/hf_tracer_utils.py [hunk 3]
```diff
@@ -3,23 +3,19 @@
...
-
     try:
         kwargs = data_gen()
         meta_args = {k: v.to('meta') for k, v in kwargs.items()}
-        graph = tracer.trace(root=model, meta_args=meta_args)
+        gm = symbolic_trace(model, meta_args=meta_args)
     except Exception as e:
         raise RuntimeError(f"Failed to trace {model.__class__.__name__}, error: {e}")
-    gm = GraphModule(model, graph, model.__class__.__name__)
     gm.recompile()
```
Comment (YuliangLiu0306) [inline tests/test_fx/test_tracer/test_hf_model/hf_tracer_utils.py:19]: This recompile call may be redundant.

File: tests/test_fx/test_tracer/test_torchrec_model/test_deepfm_model.py (hunk 2 above)
Comment (super-dainiu) [reply tests/test_fx/test_tracer/test_torchrec_model/test_deepfm_model.py:57]: Oh yes

**Pull request closed without merging**